src/cljs/kixi/nhs/exploration_ui/data.cljs

```diff
@@ -0,0 +1,56 @@
+(ns kixi.nhs.exploration-ui.data
+  (:require [om.core :as om :include-macros true]
+            [ajax.core :refer (GET POST)]
+            [clojure.walk :as walk]
+            [clojure.string :as str]))
+
+(defn date-parser [date]
+  (let [date-str (-> date (.match #"\d{4}"))]
+    (new js/Date date-str)))
```

Comment:
I think this is the bit in the next round where you will do 

```
(some #(-> date
                   (.match %)
                   ((parial new js/Date))
[#"\d{4}" #"\d{4}-\d{2}-\d{2}" ...])
```

or similiar. Health warning as usual on the syntax there and not sure if you need a try catch on the new js/Date bit.